db/channel_cache.go

```diff
@@ -48,34 +48,36 @@ type ChannelCache interface {
 	AddSkippedSequence(change *LogEntry)
 
 	// Remove purges the given doc IDs from all channel caches and returns the number of items removed.
-	Remove(collectionID uint32, docIDs []string, startTime time.Time) (count int)
+	Remove(ctx context.Context, collectionID uint32, docIDs []string, startTime time.Time) (count int)
 
 	// Returns set of changes for a given channel, within the bounds specified in options
 	GetChanges(ctx context.Context, ch channels.ID, options ChangesOptions) ([]*LogEntry, error)
 
 	// Returns the set of all cached data for a given channel (intended for diagnostic usage)
-	GetCachedChanges(ch channels.ID) ([]*LogEntry, error)
+	GetCachedChanges(ctx context.Context, ch channels.ID) ([]*LogEntry, error)
 
 	// Clear reinitializes the cache to an empty state
 	Clear()
 
 	// Size of the the largest individual channel cache, invoked for stats reporting
 	// // TODO: let the cache manage its own stats internally (maybe take an updateStats call)
-	MaxCacheSize() int
+	MaxCacheSize(context.Context) int
```

Comment:
Shouldn't we make context consistent across the full interface? Even if the current implementation doesn't actually use context in all methods?